sdk/metricsadvisor/azure-ai-metricsadvisor/azure/ai/metricsadvisor/models/_models.py

```diff
@@ -2993,3 +3006,353 @@ def _to_generated(self):
+            data_source_credential_type=self.credential_entity_type,
+            data_source_credential_name=self.name,
+            data_source_credential_description=self.description,
+            parameters=param_patch,
+        )
+
+class ServicePrincipalInKVCredentialEntity(object):
+    """ServicePrincipalCredentialEntity.
+
+    All required parameters must be populated in order to send to Azure.
+
+    :ivar credential_entity_type: Required. Type of data source credential.Constant filled by
+     server.  Possible values include: "AzureSQLConnectionString", "DataLakeGen2SharedKey",
+     "ServicePrincipal", "ServicePrincipalInKV".
+    :type credential_entity_type: str or
+     ~azure.ai.metricsadvisor.models.DataSourceCredentialType
+    :ivar id: Unique id of data source credential.
+    :vartype id: str
+    :param name: Required. Name of data source credential.
+    :type name: str
+    :keyword str description: Description of data source credential.
+    :keyword str key_vault_endpoint: Required. The Key Vault endpoint that storing the service principal.
+    :keyword str key_vault_client_id: Required. The Client Id to access the Key Vault.
+    :keyword str key_vault_client_secret: Required. The Client Secret to access the Key Vault.
+    :keyword str service_principal_id_name_in_kv: Required. The secret name of the service principal's
+     client Id in the Key Vault.
+    :keyword str service_principal_secret_name_in_kv: Required. The secret name of the service
+     principal's client secret in the Key Vault.
+    :keyword str tenant_id: Required. The tenant id of your service principal.
+    """
+
+    _attribute_map = {
+        'credential_entity_type': {'key': 'credentialEntityType', 'type': 'str'},
+        'id': {'key': 'id', 'type': 'str'},
+        'name': {'key': 'name', 'type': 'str'},
+        'description': {'key': 'description', 'type': 'str'},
+        'key_vault_endpoint': {'key': 'keyVaultEndpoint', 'type': 'str'},
+        'key_vault_client_id': {'key': 'keyVaultClientId', 'type': 'str'},
+        'key_vault_client_secret': {'key': 'keyVaultClientSecret', 'type': 'str'},
+        'service_principal_id_name_in_kv': {'key': 'servicePrincipalIdNameInKV', 'type': 'str'},
+        'service_principal_secret_name_in_kv': {'key': 'servicePrincipalSecretNameInKV', 'type': 'str'},
+        'tenant_id': {'key': 'tenantId', 'type': 'str'},
+    }
+
+    def __init__(self, name, **kwargs):
+        # type: (str, Any) -> None
+        self.credential_entity_type = 'ServicePrincipalInKV'
+        self.name = name
+        self.key_vault_endpoint = kwargs['key_vault_endpoint']
+        self.key_vault_client_id = kwargs['key_vault_client_id']
+        self.key_vault_client_secret = kwargs['key_vault_client_secret']
+        self.service_principal_id_name_in_kv = kwargs['service_principal_id_name_in_kv']
+        self.service_principal_secret_name_in_kv = kwargs['service_principal_secret_name_in_kv']
```

Comment:
Are all these keywords required? Wondering if it'd make sense to validate input and throw a ValueError if one is missing rather than surface the KeyError